hyperledger/transact-sdk-javascript
# Configure NPM publishing

This configures the NPM builds. It adds adds prepublish actions, a
modified protobuf compilation script, and an updated tsconfig to
specify a build target. It also specifies the project main and types
files in the package.json.

This does not actually publish the package to npm. That can be done by running npm publish from the project root.

To test this

- run `npm run prepublishOnly` from the project root
- run `npm link` from the project root
- create and navigate to a new empty directory outside of the project with an index.ts file in it.
- run `npm link transact-sdk-javascript` from the new project root

You should now see the types coming through in the index.ts file in your editor
